x-pack/solutions/observability/plugins/apm/server/routes/environments/has_unset_value_for_field.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { rangeQuery, termQuery } from '@kbn/observability-plugin/server';
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import { SERVICE_NAME } from '../../../common/es_fields/apm';
+import { getProcessorEventForTransactions } from '../../lib/helpers/transactions';
+import type { APMEventClient } from '../../lib/helpers/create_es_client/create_apm_event_client';
+
+/**
+ * This is used for getting the list of environments for the environment selector,
+ * filtered by range.
+ */
+
+export async function hasUnsetValueForField({
+  searchAggregatedTransactions,
+  apmEventClient,
+  serviceName,
+  fieldName,
+  start,
+  end,
+}: {
+  apmEventClient: APMEventClient;
+  searchAggregatedTransactions: boolean;
+  serviceName: string | undefined;
+  fieldName: string;
+  start: number;
+  end: number;
+}): Promise<boolean> {
+  const params = {
+    apm: {
+      events: [
+        getProcessorEventForTransactions(searchAggregatedTransactions),
+        ProcessorEvent.metric,
+        ProcessorEvent.error,
+      ],
+    },
+    track_total_hits: true,
+    size: 0,
```

Comment:
Are we interested in whether at least one service without `service.environment` exists? If so, this might be more performant

```ts
terminate_after: 1, // stops when one item is found
track_total_hits: true,
size: 0,
```